xc/device.ml

```diff
@@ -605,8 +605,9 @@ let qemu_media_change ~xs device _type params =
 	Xs.transaction xs (fun t -> t.Xst.writev backend_path back_delta);
 	debug "Media changed: params = %s" pathtowrite;
 
-	try
-		ignore(with_xs (fun xs -> xs.Xs.read (Printf.sprintf "/libxl/%d/dm-version" device.frontend.domid)) = "qemu_xen");
+
+	if is_upstream_qemu device.frontend.domid
+	then begin
 		let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" device.frontend.domid) in
```

Comment:
This same string and the rest of the code is being used every time you need to connect to `qmp-libxl`, it may be worth to extract it out to an helper. Something like 

```ocaml 
let qmp_write domit cmd =
  let c = Qmp_protocol.connect (Printf.sprintf "/var/run/xen/qmp-libxl-%d" domid) in
￼  Qmp_protocol.negotiate c;
  Qmp_protocol.write c cmd;
  Qmp_protocol.close c
```

Maybe in a `try...with` or a `finally` that deals with potential errors.